File: typescript/fizzbuzz.ts
function beatit(num:number,beat:number):boolean {
    return (num % beat == 0);
}
function fizzbuzz(num: number): string {
    let beat = "";
    if (beatit(num, 3))
        beat += "Fizz";
    if (beatit(num, 5))
        beat += "Buzz";
    if (beat == "")
        return "Nah!!";
    else return beat;
}

function main() {
    for (let num = 1; num <= 100; num++){
        let rythm = fizzbuzz(num);
        if (rythm == "Nah!!")
            console.log(num);
        else
            console.log(rythm);
            
        
    }
}
main();

File: R/fizzbuzz.r
fizzbuzz <- function(num) {
    if (num %% 3 == 0 && num %% 5 == 0) {
        print("FizzBuzz")
    } else if (num %% 3 == 0) {
        print("Fizz")
    } else if (num %% 5 == 0) {
        print("Buzz")
    } else {
        print(num)
    }
    
}
sapply(1:100, function(x) fizzbuzz(x))